### PR TITLE
Add test for ActiveSamplingRequest

### PR DESCRIPTION
**Pull request by ackintosh**
## Issue Addressed

I've implemented batching sampling requests in #6256, but there are no unit tests for it yet.

## Proposed Changes

Add a test for ActiveSamplingRequest to ensure that the requests are batched, and check the status of the requests.

### CODE REVIEW

**Comment by michaelsproul** [comment]
Please update to point at `unstable` by either:

1. Rebasing on `unstable` (if your branch has a small number of commits that are easy to tease out), or
2. Merging `origin/das` into this PR: `git fetch origin; git merge origin/das`. This will result in the smallest number of conflict resolutions and is better for branches that already contain merge commits or have extensive history.

**Comment by ackintosh** [comment]
Thanks for the comment Michael! I've rebased this on `unstable`.

File: beacon_node/network/src/sync/sampling.rs [hunk 1]
```diff
@@ -626,3 +635,257 @@ mod request {
...
+                    signature: Signature::empty(),
+                },
+                kzg_commitments_inclusion_proof: FixedVector::default(),
+            }));
+        }
+        (list, Duration::default())
+    }
+
+    #[test]
+    fn test_active_sampling_request() {
```
Comment (dapplion) [inline beacon_node/network/src/sync/sampling.rs]: Can you write tests in this format and file?  https://github.com/sigp/lighthouse/blob/5a966874da03cd987fef1629f732898a2859ce87/beacon_node/network/src/sync/block_lookups/tests.rs#L2011

We recently avoided adding unit tests too close to impl for event-based tests

**Comment by ackintosh** [comment]
Thanks @dapplion! I've written tests in the tests.rs file.

**Review by jimmygchen — APPROVED** [review]
Nice test, very tidy 🎉

**Comment by jimmygchen** [comment]
@mergify queue

**Pull request merged**